ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/numberinput/v1/numberinput/clientlibs/site/js/numberinputview.js

```diff
@@ -63,11 +63,24 @@
             return this.element.querySelector(NumberInput.selectors.qm);
         }
 
+        _updateValue(value) {
+            if (this.widgetObject) {
+                this.widgetObject.setValue(value);
+            } else {
+                super._updateValue(value);
+            }
+        }
+
         setModel(model) {
             super.setModel(model);
-            this.widget.addEventListener('blur', (e) => {
-                this._model.value = e.target.value;
-            })
+            // only initialize if patterns are set
+            if (this._model._jsonModel.editFormat || this._model._jsonModel.displayFormat) {
```

Comment:
there should be no need for the view to refer to _jsonModel. ideally, some getter at Field level should be there in model. something like `this._model.editFormat`